components/input/input-otp.component.ts

```diff
@@ -0,0 +1,232 @@
+  ReactiveFormsModule,
+  Validators
+} from '@angular/forms';
+import { takeUntil, tap } from 'rxjs/operators';
+
+import { NzDestroyService } from 'ng-zorro-antd/core/services';
+import { NzSafeAny, NzSizeLDSType, NzStatus, OnTouchedType } from 'ng-zorro-antd/core/types';
+
+import { NzInputDirective } from './input.directive';
+
+@Component({
+  selector: 'nz-input-otp',
+  exportAs: 'nzInputOtp',
+  preserveWhitespaces: false,
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    @for (item of otpArray.controls; track $index) {
+      <input
+        nz-input
+        class="ant-otp-input"
+        type="text"
+        maxlength="1"
+        size="1"
+        [nzSize]="nzSize"
+        [formControl]="item"
+        [nzStatus]="nzStatus"
+        (input)="onInput($index, $event)"
+        (focus)="onFocus($event)"
+        (keydown)="onKeyDown($index, $event)"
+        (paste)="onPaste($index, $event)"
+        #otpInput
+      />
+    }
+  `,
+  host: {
+    class: 'ant-otp'
+  },
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => NzInputOtpComponent),
+      multi: true
+    },
+    NzDestroyService
+  ],
+  imports: [NzInputDirective, ReactiveFormsModule],
+  standalone: true
+})
+export class NzInputOtpComponent implements ControlValueAccessor, OnChanges {
+  @ViewChildren('otpInput') otpInputs!: QueryList<ElementRef>;
+
+  @Input({ transform: numberAttribute }) nzLength: number = 6;
+  @Input() nzSize: NzSizeLDSType = 'default';
+  @Input({ transform: booleanAttribute }) disabled = false;
+  @Input() nzStatus: NzStatus = '';
+  @Input() nzFormatter: (value: string) => string = value => value;
+  @Input() nzMask: string | null = null;
+
+  protected otpArray!: FormArray<FormControl<string>>;
+  private internalValue: string[] = [];
+  private onChangeCallback?: (_: NzSafeAny) => void;
+  onTouched: OnTouchedType = () => {};
+
+  constructor(
+    private readonly formBuilder: FormBuilder,
+    private readonly nzDestroyService: NzDestroyService
+  ) {
+    this.createFormArray();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['nzLength']?.currentValue) {
+      this.createFormArray();
+    }
+
+    if (changes['disabled']) {
+      this.setDisabledState(this.disabled);
+    }
+  }
+
+  onInput(index: number, event: Event): void {
+    const inputElement = event.target as HTMLInputElement;
+    const nextInput = this.otpInputs.toArray()[index + 1];
+
+    if (inputElement.value && nextInput) {
+      nextInput.nativeElement.focus();
+    } else if (!nextInput) {
+      this.selectInputBox(index);
+    }
+  }
+
+  onFocus(event: FocusEvent): void {
+    const inputElement = event.target as HTMLInputElement;
+    inputElement.select();
+  }
+
+  onKeyDown(index: number, event: KeyboardEvent): void {
+    const previousInput = this.otpInputs.toArray()[index - 1];
+
+    if (event.key === 'Backspace') {
```

Comment:
```ts
import { BACKSPACE } from '@angular/cdk/keycodes';
```
```suggestion
    if (event.keyCode === BACKSPACE) {
```

Reply:
u sure?
keyCode is deprecated though...